src/main/kotlin/com/exactpro/th2/check1/rule/check/CheckRuleTask.kt

```diff
@@ -46,19 +46,25 @@ class CheckRuleTask(
     eventBatchRouter: MessageRouter<EventBatch>
 ) : AbstractCheckTask(ruleConfiguration, startTime, sessionKey, parentEventID, messageStream, eventBatchRouter) {
 
-    private var protoMessageFilter: RootMessageFilter? = protoMessageFilter
-
-    private var messageFilter: SailfishFilter? = SailfishFilter(
-        CONVERTER.fromProtoPreFilter(protoMessageFilter),
-        protoMessageFilter.toCompareSettings()
+    private class Refs(
+        val protoMessageFilter: RootMessageFilter,
+        val messageFilter: SailfishFilter,
+        val metadataFilter: SailfishFilter?
     )
-
-    private var metadataFilter: SailfishFilter? = protoMessageFilter.metadataFilterOrNull()?.let {
-        SailfishFilter(
-            CONVERTER.fromMetadataFilter(it, METADATA_MESSAGE_NAME),
-            it.toComparisonSettings()
-        )
-    }
+    private var _refs: Refs? = Refs(
+        protoMessageFilter = protoMessageFilter,
+        messageFilter = SailfishFilter(
+            CONVERTER.fromProtoPreFilter(protoMessageFilter),
+            protoMessageFilter.toCompareSettings()
+        ),
+        metadataFilter = protoMessageFilter.metadataFilterOrNull()?.let {
+            SailfishFilter(
+                CONVERTER.fromMetadataFilter(it, METADATA_MESSAGE_NAME),
+                it.toComparisonSettings()
+            )
+        }
+    )
+    private val refs get() = _refs ?: throw IllegalStateException("Requesting references after references has been removed")
```

Comment:
You can use `error` method from Kotlin. It throws the same type of the exception
```suggestion
    private val refs get() = _refs ?: error("Requesting references after references has been removed")
```

Reply:
You didn't change the code here